### DISCUSSION TITLE
Cannot enable automerge

### DISCUSSION THREAD

**Discussion by cloudymax** [renovatebot/renovate#25420] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Hoi, we are trying to install and use the bot on a github repo, however we are unable to get the bot to enable automerge.

## Links

Real Repo: https://github.com/small-hack/argocd-apps
Minimal Debug Repo: https://github.com/small-hack/renovatebot-argo

## What we tried

Configs we have tried that do not work:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base"
  ],
  "argocd": {
    "fileMatch": ["\\.yaml$"]
  },
  "packageRules": [
    { 
      "matchManagers": ["argocd"],
      "matchUpdateTypes": ["patch"],
      "matchCurrentVersion": "!/^0/",
      "automerge": true
    }
  ]
}
```

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended"
  ],
  "argocd": {
    "fileMatch": ["\\.yaml$"]
  },
  "packageRules": [
    { 
      "matchManagers": ["argocd"],
      "matchUpdateTypes": ["patch"],
      "matchCurrentVersion": "!/^0/",
      "automerge": true
    }
  ]
}
```

```json
{
  "argocd": {
    "fileMatch": ["\\.yaml$"]
  },
  "packageRules": [
    {
      "matchManagers": ["argocd"],
      "matchUpdateTypes": ["patch"],
      "matchCurrentVersion": "!/^0/",
      "automerge": true
    }
  ]
}
```
The configurations are nearly copy-paste from the docs, and the repo has lint/validation CI jobs which all return as passing.

Each config was tested with `renovate-config-validator` and found to be valid at the time of its creation. They were merged via a new branch and PR as directed in the documents.

Each time the config was changed, the bot was given several days to show any change in behavior but regardless of what config we use the automerge option remains disabled and old PR's are never re-evaluated. Clicking the box to rebase and re-try does not trigger the bot to use the new configuration.

<img width="2000" alt="Screenshot 2023-10-25 at 15 48 35" src="https://github.com/renovatebot/renovate/assets/84841307/94e8e94c-e5b7-4fee-b615-572e14ab9d40">

## Remaining Questions

- How can we get automerge to work given that `"automerge": true` seems to have no effect?

- How can we re-create or test PRs opened by the bot once the config file has been changed?

- How can we test configurations for correctness if the validation tool is not catching improper configurations?

Thanks and Cheers!

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
The reproduction repo contains a major update only

**Reply by cloudymax** [reply]
Thanks for pointing that out - I've swapped the project out for one that is only a patch update for correctness and the stated behavior can now be seen in the new PR from the bot:

<img width="551" alt="Screenshot 2023-10-25 at 17 47 49" src="https://github.com/renovatebot/renovate/assets/84841307/e7ea6b8b-5777-431f-b35c-4f2b6db929e2">

**Reply by rarkins** [reply, marked as answer]
You have disabled 0.x automerging:

https://github.com/small-hack/renovatebot-argo/blob/eecdbbd43b122cc4ade5bdcc30c99f0a07c5a0a9/renovate.json#L9

https://github.com/small-hack/renovatebot-argo/pull/4

**Reply by cloudymax** [reply]
Thanks so much for the help!
